packages-exp/messaging-exp/src/helpers/register.ts

```diff
@@ -41,8 +45,32 @@ const messagingFactory: InstanceFactory<'messaging-exp'> = (
   );
 };
 
-export function registerMessaging(): void {
+const SwMessagingFactory: InstanceFactory<'messaging-exp'> = (
+  container: ComponentContainer
+) => {
+  // Top-level 'await' requires 'module' option set to 'esnext' or 'system', and 'target' option
+  // set to 'es2017' or higher. For compatibility, use async expression here.
```

Comment:
Please add the comment since it's not at top level now. We still want to use the iife pattern here because the factory needs to be sync.